src/rewriter/basic_rewrite_rcons.cpp

```diff
@@ -63,6 +65,18 @@ bool BasicRewriteRCons::prove(
     Trace("trewrite-rcons") << "...EXISTS_ELIM" << std::endl;
     return true;
   }
+
+  if (tid == theory::THEORY_BV)
```

Comment:
We should only try this after reconstruction fails.  Let's add a new method `BasicRewriteRCons::postProve` which is called by the main proof reconstruction only when it fails and move this block to this method.

Reply:
done